pygsti/report/workspaceplots.py

```diff
@@ -1841,7 +2007,15 @@ def _create(self, plottypes, circuits, dataset, model, prec, sum_up, box_labels,
             else: assert(False), "Internal logic error"  # pragma: no cover
 
             if typ == "boxes":
-                newfig = _circuit_color_boxplot(circuit_struct, subMxs, colormap,
+                if not isinstance(circuit_struct, _PlaquetteGridCircuitStructure):
+                    #if not a plaquette structure then maybe try returning a NotApplicable object
+                    #for the figure?
+                    return NotApplicable(self.ws)
+                else:
+                    #I am expecting this cast won't do anything at the moment, but
```

Comment:
I don't understand this section. Does it not do anything now? What is it intended to do in the future then? Why doesn't it work now?

Reply:
Good question. As far as I can recall what I was noting in that comment is that in the original implementation of this method there are a number of places where we cast circuits to a PlaquetteGridCircuitStructure even though we've already checked for whether the object being cast was an instance of one earlier (as here). In a nutshell the comment can be translated as: 'I'm not sure why it is exactly we're doing these casts, but @enielse probably had a reason (maybe some future code extensions) and I don't see any harm in keeping them just to be safe'.